pkgsend: Action payload 'LICENCE.md' was not found in any of the provided locations:
/export/home/andreas/oi-userland/components/network/megasync/build/prototype/i386/mangled
/export/home/andreas/oi-userland/components/network/megasync/build/prototype/i386
/export/home/andreas/oi-userland/components/network/megasync/build
/export/home/andreas/oi-userland/components/network/megasync
MEGAsync-v4.6.3.0_Linux
gmake: *** [/export/home/andreas/oi-userland/make-rules/ips.mk:555: /export/home/andreas/oi-userland/components/network/megasync/build/manifest-i386-megasync.published] Error 1

--- a/components/network/megasync/LICENCE.md
+++ b/components/network/megasync/LICENCE.md
@@ -1,0 +1,107 @@
+MEGA LIMITED CODE REVIEW LICENCE
+================================
+
+This licence grants you the rights, and only the rights, set out
+below, to access and review Mega's code. If you take advantage of
+these rights, you accept this licence. If you do not accept the
+licence, do not access the code.
+
+Words used in the Mega Limited Terms of Service
+[https://mega.nz/terms] have the same meaning in this licence. Where
+there is any inconsistency between this licence and those Terms of
+Service, these terms prevail.
+
+
+Section 1 -- Definitions
+------------------------
+
+- "code" means the Mega code made available from time, in our sole
+  discretion, for access under this licence at [https://github.com/].
+  Reference to code in this licence means the code and any part of it
+  and any derivative of it.
+
+- "defect" means a defect, bug, backdoor, security issue or other
+  deficiency in the code.
+
+- "review " means to access, analyse, test and otherwise review the
+  code as a reference, for the sole purpose of analysing it for
+  defects.
+
+- "you" means the licensee of rights set out in this licence.
+
+
+Section 2 -- Grant of Rights
+----------------------------
+
+1. Subject to the terms of this licence, we grant you a
+   non-transferable, non-exclusive, worldwide, royalty-free licence to
+   access and use the code solely for review purposes.
+
+2. You may provide the code to anyone else and publish excerpts of it
+   for the purposes of review and commentary, provided that when you
+   do so you make any recipient of the code aware of the terms of this
+   licence and that you attribute the code to Mega.
+
+3. Other than in respect of those parts of the code that were
+   developed by other parties and as specified strictly in accordance
+   with the open source and other licences under which those parts of
+   the code have been made available, as set out on our website or in
+   those items of code, you are not entitled to use or do anything
+   with the code for any commercial or other purpose, other than
+   review and commentary on it.
+
+4. Subject to the terms of this licence, you must at all times comply
+   with and shall be bound by our Terms of Use
+   [https://mega.nz/terms], Privacy Policy [https://mega.nz/privacy]
+   and Takedown Guidance Policy [https://mega.nz/takedown].
+
+
+Section 3 -- Limitations
+------------------------
+
+1. This licence does not grant you any rights to use Mega's name,
+   logo, or trademarks and you must not in any way indicate you are
+   authorised to speak on behalf of Mega.
+
+2. If you issue proceedings in any jurisdiction against Mega because
+   you consider Mega has infringed copyright or any patent right in
+   respect of the code (including any joinder or counterclaim), your
+   licence to the code is automatically terminated.
+
+3. THE CODE IS MADE AVAILABLE "AS-IS" AND WITHOUT ANY EXPRESS OF
+   IMPLIED GUARANTEES AS TO FITNESS, MERCHANTABILITY, NON-INFRINGEMENT
+   OR OTHERWISE. IT IS NOT BEING PROVIDED IN TRADE BUT ON A VOLUNTARY
+   BASIS ON OUR PART AND YOURS AND IS NOT MADE AVAILABE FOR CONSUMER
+   USE OR ANY OTHER USE OUTSIDE THE TERMS OF THIS LICENCE. ANYONE
+   ACCESSING THE CODE SHOULD HAVE THE REQUISITE EXPERTISE TO SECURE
+   THEIR OWN SYSTEM AND DEVICES AND TO ACCESS AND USE THE CODE FOR
+   REVIEW PURPOSES. YOU BEAR THE RISK OF ACCESSING AND USING IT. IN
+   PARTICULAR, MEGA BEARS NO LIABILITY FOR ANY INTERFERENCE WITH OR
+   ADVERSE EFFECT ON YOUR SYSTEM OR DEVICES AS A RESULT OF YOUR
+   ACCESSING AND USING THE CODE.
+
+
+Section 4 -- Termination, suspension and variation
+--------------------------------------------------
+
+1. We may suspend, terminate or vary the terms of this licence and any
+   access to the code at any time, without notice, for any reason or
+   no reason, in respect of any licensee, group of licensees or all
+   licensees.
+
+
+Section 5 -- General
+--------------------
+
+1. This licence and its interpretation and operation are governed
+   solely by New Zealand law. We and each you submit to the exclusive
+   jurisdiction of the New Zealand arbitral tribunals as further
+   described in our Terms of Service and you agree not to raise any
+   jurisdictional issue if we need to enforce an arbitral award or
+   judgment in New Zealand or another country.
+
+2. Questions and comments regarding this licence are welcomed and
+   should be addressed to [support@mega.nz].
+
+
+Last updated 20 January 2016.

--- a/components/network/megasync/Makefile
+++ b/components/network/megasync/Makefile
@@ -1,0 +1,107 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2022, Daniel Chan
+#
+
+BUILD_BITS=			64
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		MEGAsync
+COMPONENT_VERSION=	4.6.3
+_REL=Linux
+COMPONENT_PREP_GIT=yes
+GIT_TAG=v$(COMPONENT_VERSION).0_$(_REL)
+COMPONENT_SUMMARY=	Easy automated syncing between your computers and your MEGA cloud drive
+COMPONENT_SRC=		$(COMPONENT_NAME)-$(GIT_TAG)
+COMPONENT_PROJECT_URL=	https://github.com/meganz/MEGAsync
+GIT_REPO=https://github.com/meganz/MEGAsync.git
+GIT_COMMIT_ID=tags/$(GIT_TAG)
+SDK_COMMIT_ID=995b5c9b599ae2e0b72ef0029775bd6c9a3c4326
+SDK_REPO=https://github.com/meganz/sdk.git
+COMPONENT_ARCHIVE_SRC=git
+COMPONENT_FMRI=	network/megasync
+COMPONENT_CLASSIFICATION= Applications/System Utilities
+COMPONENT_LICENSE=	custom:MEGA LIMITED CODE REVIEW LICENCE
+COMPONENT_LICENSE_FILE=	LICENCE.md
+
+include $(WS_MAKE_RULES)/common.mk
+COMPONENT_POST_UNPACK_ACTION= \
+	git clone $(SDK_REPO) $(SOURCE_DIR)/src/MEGASync/mega; \
+	cd $(SOURCE_DIR)/src/MEGASync/mega && git checkout $(SDK_COMMIT_ID)
+CONFIGURE_OPTIONS += with_pcre=/usr
+CONFIGURE_SCRIPT = configure
+COMPONENT_PRE_CONFIGURE_ACTION= \
+	(mv $(SOURCE_DIR)/* $(@D)); \
+	cd $(@D)/src/MEGASync/mega && ./autogen.sh
+COMPONENT_PRE_BUILD_ACTION= \
+	cd $(@D)/src && qmake "CONFIG += FULLREQUIREMENTS" "QMAKE_LFLAGS += -Wl,-z,nodefs -Wl,-z,ignore -Wl,-z,redlocsym -fPIC" "QMAKE_CC = $(CC)" "QMAKE_CXX = $(CXX)" MEGA.pro  && lrelease MEGASync/MEGASync.pro
+COMPONENT_POST_BUILD_ACTION= \
+	mv $(BUILD_DIR)/amd64/LICENCE.md $(SOURCE_DIR)/../LICENCE.md
+COMPONENT_INSTALL_ARGS += INSTALL_ROOT=$(PROTO_DIR)
+COMPONENT_POST_INSTALL_ACTION = \
+	cd $(@D)/src/ && mkdir -p $(PROTO_DIR)/usr/bin/ && ginstall -Dm 755 MEGASync/megasync $(PROTO_DIR)/usr/bin/
+
+$(BUILD_DIR)/%/.configured:	$(SOURCE_DIR)/.prep
+	($(RM) -rf $(@D) ; $(MKDIR) $(@D))
+	$(COMPONENT_PRE_CONFIGURE_ACTION)
+	(cd $(@D) ; cd $(@D)/src/MEGASync/mega && $(ENV) $(CONFIGURE_ENV) $(CONFIG_SHELL) \
+		$(CONFIGURE_SCRIPT) $(CONFIGURE_OPTIONS))
+	$(COMPONENT_POST_CONFIGURE_ACTION)
+	$(TOUCH) $@
+
+$(BUILD_DIR)/%/.built:	$(BUILD_DIR)/%/.configured
+	$(COMPONENT_PRE_BUILD_ACTION)
+	(cd $(@D) ; cd $(@D)/src && $(ENV) $(COMPONENT_BUILD_ENV) \
+		$(GMAKE) $(COMPONENT_BUILD_GMAKE_ARGS) $(COMPONENT_BUILD_ARGS) \
+		$(COMPONENT_BUILD_TARGETS))
+	$(COMPONENT_POST_BUILD_ACTION)
+ifeq   ($(strip $(PARFAIT_BUILD)),yes)
+	-$(PARFAIT) build
+endif
+	$(TOUCH) $@
+
+$(BUILD_DIR)/%/.installed:	$(BUILD_DIR)/%/.built
+	$(COMPONENT_PRE_INSTALL_ACTION)
+	(cd $(@D) ; cd $(@D)/src && $(ENV) $(COMPONENT_INSTALL_ENV) $(GMAKE) \
+			$(COMPONENT_INSTALL_ARGS) $(COMPONENT_INSTALL_TARGETS))
+	$(COMPONENT_POST_INSTALL_ACTION)
+	$(TOUCH) $@
+
+# Auto-generated dependencies
+REQUIRED_PACKAGES += $(GCC_RUNTIME_PKG)
+REQUIRED_PACKAGES += $(GXX_RUNTIME_PKG)
+REQUIRED_PACKAGES += database/sqlite-3
+REQUIRED_PACKAGES += library/cryptopp
+REQUIRED_PACKAGES += library/libcares
+REQUIRED_PACKAGES += library/libfreeimage
+REQUIRED_PACKAGES += library/libmediainfo
+REQUIRED_PACKAGES += library/libuv
+REQUIRED_PACKAGES += library/libzen
+REQUIRED_PACKAGES += library/qt5
+REQUIRED_PACKAGES += library/security/libsodium
+REQUIRED_PACKAGES += library/security/openssl
+REQUIRED_PACKAGES += library/zlib
+REQUIRED_PACKAGES += system/library
+REQUIRED_PACKAGES += system/library/math
+REQUIRED_PACKAGES += web/curl
+REQUIRED_PACKAGES += x11/library/libxcb

--- a/components/network/megasync/manifests/sample-manifest.p5m
+++ b/components/network/megasync/manifests/sample-manifest.p5m
@@ -1,0 +1,45 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2022 <contributor>
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/bin/megasync
+file path=usr/share/applications/megasync.desktop
+file path=usr/share/doc/megasync/distro
+file path=usr/share/doc/megasync/version
+file path=usr/share/icons/hicolor/128x128/apps/mega.png
+file path=usr/share/icons/hicolor/16x16/apps/mega.png
+file path=usr/share/icons/hicolor/256x256/apps/mega.png
+file path=usr/share/icons/hicolor/32x32/apps/mega.png
+file path=usr/share/icons/hicolor/48x48/apps/mega.png
+file path=usr/share/icons/hicolor/scalable/status/megaalert.svg
+file path=usr/share/icons/hicolor/scalable/status/megalogging.svg
+file path=usr/share/icons/hicolor/scalable/status/megapaused.svg
+file path=usr/share/icons/hicolor/scalable/status/megasynching.svg
+file path=usr/share/icons/hicolor/scalable/status/megauptodate.svg
+file path=usr/share/icons/hicolor/scalable/status/megawarning.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megaalert.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megalogging.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megapaused.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megasynching.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megauptodate.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megawarning.svg

--- a/components/network/megasync/megasync.p5m
+++ b/components/network/megasync/megasync.p5m
@@ -1,0 +1,45 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2022 Daniel Chan
+#
+
+set name=pkg.fmri value=pkg:/$(COMPONENT_FMRI)@$(IPS_COMPONENT_VERSION),$(BUILD_VERSION)
+set name=pkg.summary value="$(COMPONENT_SUMMARY)"
+set name=info.classification value="$(COMPONENT_CLASSIFICATION)"
+set name=info.upstream-url value=$(COMPONENT_PROJECT_URL)
+set name=info.source-url value=$(COMPONENT_ARCHIVE_URL)
+set name=org.opensolaris.consolidation value=$(CONSOLIDATION)
+
+license $(COMPONENT_LICENSE_FILE) license='$(COMPONENT_LICENSE)'
+
+file path=usr/bin/megasync
+file path=usr/share/applications/megasync.desktop
+file path=usr/share/doc/megasync/distro
+file path=usr/share/doc/megasync/version
+file path=usr/share/icons/hicolor/128x128/apps/mega.png
+file path=usr/share/icons/hicolor/16x16/apps/mega.png
+file path=usr/share/icons/hicolor/256x256/apps/mega.png
+file path=usr/share/icons/hicolor/32x32/apps/mega.png
+file path=usr/share/icons/hicolor/48x48/apps/mega.png
+file path=usr/share/icons/hicolor/scalable/status/megaalert.svg
+file path=usr/share/icons/hicolor/scalable/status/megalogging.svg
+file path=usr/share/icons/hicolor/scalable/status/megapaused.svg
+file path=usr/share/icons/hicolor/scalable/status/megasynching.svg
+file path=usr/share/icons/hicolor/scalable/status/megauptodate.svg
+file path=usr/share/icons/hicolor/scalable/status/megawarning.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megaalert.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megalogging.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megapaused.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megasynching.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megauptodate.svg
+file path=usr/share/icons/ubuntu-mono-dark/status/24/megawarning.svg

--- a/components/network/megasync/patches/01-configure.ac.patch
+++ b/components/network/megasync/patches/01-configure.ac.patch
@@ -1,0 +1,59 @@
+--- MEGAsync_ori/src/MEGASync/mega/configure.ac	Wed Mar  9 01:46:53 2022
++++ MEGAsync_patched/src/MEGASync/mega/configure.ac	Mon Feb 28 22:19:04 2022
+@@ -612,26 +612,26 @@
+    *)
+ 
+     # determine if library is installed
+-    if test -d "$with_pcre/lib64"; then
+-        LDFLAGS="-L$with_pcre/lib64 $LDFLAGS"
+-        CXXFLAGS="-I$with_pcre/include $CXXFLAGS"
+-        CPPFLAGS="-I$with_pcre/include $CPPFLAGS"
++    if test -d "$with_pcre/lib/amd64"; then
++        LDFLAGS="-L$with_pcre/lib/amd64 $LDFLAGS"
++        CXXFLAGS="-I$with_pcre/include/pcre $CXXFLAGS"
++        CPPFLAGS="-I$with_pcre/include/pcre $CPPFLAGS"
+ 
+         AC_CHECK_HEADERS(pcrecpp.h,
+-         PCRE_CXXFLAGS="-I$with_pcre/include"
+-         PCRE_CPPFLAGS="-I$with_pcre/include"
+-         PCRE_LDFLAGS="-L$with_pcre/lib64",
++         PCRE_CXXFLAGS="-I$with_pcre/include/pcre"
++         PCRE_CPPFLAGS="-I$with_pcre/include/pcre"
++         PCRE_LDFLAGS="-L$with_pcre/lib/amd64",
+          AC_MSG_ERROR([pcrecpp.h header not found or not usable])
+          )
+     else
+-        LDFLAGS="-L$with_pcre/lib32 $LDFLAGS"
+-        CXXFLAGS="-I$with_pcre/include $CXXFLAGS"
+-        CPPFLAGS="-I$with_pcre/include $CPPFLAGS"
++        LDFLAGS="-L$with_pcre/lib $LDFLAGS"
++        CXXFLAGS="-I$with_pcre/include/pcre $CXXFLAGS"
++        CPPFLAGS="-I$with_pcre/include/pcre $CPPFLAGS"
+ 
+         AC_CHECK_HEADERS(pcrecpp.h,
+-         PCRE_CXXFLAGS="-I$with_pcre/include"
+-         PCRE_CPPFLAGS="-I$with_pcre/include"
+-         PCRE_LDFLAGS="-L$with_pcre/lib32",
++         PCRE_CXXFLAGS="-I$with_pcre/include/pcre"
++         PCRE_CPPFLAGS="-I$with_pcre/include/pcre"
++         PCRE_LDFLAGS="-L$with_pcre/lib",
+          AC_MSG_ERROR([pcrecpp.h header not found or not usable])
+          )
+     fi
+@@ -1652,13 +1652,13 @@
+    *)
+ 
+     # determine if library is installed
+-    if test -d "$with_freeimage/lib"; then
+-        LDFLAGS="-L$with_freeimage/lib $LDFLAGS"
++    if test -d "$with_freeimage/lib/amd64"; then
++        LDFLAGS="-L$with_freeimage/lib/amd64 $LDFLAGS"
+         CXXFLAGS="-I$with_freeimage/include $CXXFLAGS"
+         CPPFLAGS="-I$with_freeimage/include $CPPFLAGS"
+ 
+         AC_CHECK_HEADERS([FreeImage.h],[
+-         FI_LDFLAGS="-L$with_freeimage/lib"
++         FI_LDFLAGS="-L$with_freeimage/lib/amd64"
+          FI_CXXFLAGS="-I$with_freeimage/include"
+          FI_CPPFLAGS="-I$with_freeimage/include"],
+          AC_MSG_ERROR([FreeImage.h header not found or not usable])

--- a/components/network/megasync/patches/02-MEGASync.pro.patch
+++ b/components/network/megasync/patches/02-MEGASync.pro.patch
@@ -1,0 +1,29 @@
+--- MEGAsync_ori/src/MEGASync/MEGASync.pro	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/MEGASync.pro	Sat Mar  5 15:51:17 2022
+@@ -91,7 +91,7 @@
+ DEFINES += NOMINMAX
+ }
+ else:contains(BUILD_ARCH, x86_64) { #Notice this might not work for clang!
+-CONFIG += USE_PDFIUM
++# CONFIG += USE_PDFIUM
+ }
+ 
+ unix:!macx {
+@@ -104,7 +104,7 @@
+ }
+ 
+ # Drive notifications (for SDK)
+-CONFIG += USE_DRIVE_NOTIFICATIONS
++# CONFIG += USE_DRIVE_NOTIFICATIONS
+ 
+ include(gui/gui.pri)
+ include(mega/bindings/qt/sdk.pri)
+@@ -215,7 +215,7 @@
+ DEFINES += REQUIRE_HAVE_PDFIUM
+ }
+ else:contains(BUILD_ARCH, x86_64) { #Notice this might not work for clang!
+-DEFINES += REQUIRE_HAVE_PDFIUM
++# DEFINES += REQUIRE_HAVE_PDFIUM
+ }
+ 
+ #DEFINES += REQUIRE_ENABLE_CHAT

--- a/components/network/megasync/patches/03-ProxySettings.cpp.patch
+++ b/components/network/megasync/patches/03-ProxySettings.cpp.patch
@@ -1,0 +1,38 @@
+--- MEGAsync_ori/src/MEGASync/gui/ProxySettings.cpp	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/gui/ProxySettings.cpp	Sat Mar  5 15:36:25 2022
+@@ -33,7 +33,7 @@
+     connect(mUi->rProxyManual, &QRadioButton::clicked, this, [this]{setManualMode(true);});
+     connect(mUi->cProxyRequiresPassword, &QCheckBox::toggled, this, [this]{setManualMode(true);});
+     connect(mUi->rNoProxy, &QRadioButton::clicked, this, [this]{setManualMode(false);});
+-#ifndef Q_OS_LINUX
++#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
+     connect(mUi->rProxyAuto, &QRadioButton::clicked, this, [this]{setManualMode(false);});
+ #endif
+ }
+@@ -48,7 +48,7 @@
+ void ProxySettings::initialize()
+ {
+     mUi->rNoProxy->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_NONE);
+-#ifndef Q_OS_LINUX
++#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
+     mUi->rProxyAuto->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_AUTO);
+ #endif
+     mUi->rProxyManual->setChecked(mPreferences->proxyType() == Preferences::PROXY_TYPE_CUSTOM);
+@@ -99,7 +99,7 @@
+     {
+         mPreferences->setProxyType(Preferences::PROXY_TYPE_NONE);
+     }
+-#ifndef Q_OS_LINUX
++#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
+     else if (mUi->rProxyAuto->isChecked())
+     {
+         mPreferences->setProxyType(Preferences::PROXY_TYPE_AUTO);
+@@ -147,7 +147,7 @@
+             proxy.setPassword(mUi->eProxyPassword->text());
+         }
+     }
+-#ifndef Q_OS_LINUX
++#if !defined(Q_OS_LINUX) && !defined(Q_OS_SOLARIS)
+     else if (mUi->rProxyAuto->isChecked())
+     {
+         MegaProxy *proxySettings = mApp->getMegaApi()->getAutoProxySettings();

--- a/components/network/megasync/patches/04-utils.cpp.patch
+++ b/components/network/megasync/patches/04-utils.cpp.patch
@@ -1,0 +1,11 @@
+--- MEGAsync_ori/src/MEGASync/mega/src/utils.cpp	Wed Mar  9 22:04:59 2022
++++ MEGASync_patched/mega/src/utils.cpp	Sat Mar  5 15:44:11 2022
+@@ -1916,7 +1916,7 @@
+ #error "localtime is not thread safe in this compiler; please use a later one"
+ #else //POSIX
+     t = mktime(&dst);
+-    t += dst.tm_gmtoff - dst.tm_isdst * 3600;
++    t += dst.tm_isdst * 3600;
+ #endif
+     return t;
+ }

--- a/components/network/megasync/patches/05-fs.cpp.patch
+++ b/components/network/megasync/patches/05-fs.cpp.patch
@@ -1,0 +1,106 @@
+--- MEGAsync_ori/src/MEGASync/mega/src/posix/fs.cpp	Thu Mar 10 00:45:05 2022
++++ MEGAsync_patched/src/MEGASync/mega/src/posix/fs.cpp	Sat Mar  5 15:48:56 2022
+@@ -39,6 +39,10 @@
+ #include <uuid/uuid.h>
+ #endif
+ 
++#ifdef __solaris__
++#include <sys/statvfs.h>
++#endif
++
+ #ifdef __linux__
+ 
+ #ifndef __ANDROID__
+@@ -1276,6 +1280,7 @@
+     dirent* d;
+     int removed;
+     struct stat statbuf;
++	struct stat s;
+     PosixFileSystemAccess pfsa;
+ #ifdef USE_IOS
+     const string namestr = adjustBasePath(name);
+@@ -1303,7 +1308,8 @@
+ 
+             while ((d = readdir(dp)))
+             {
+-                if (d->d_type != DT_DIR
++				stat(d->d_name, &s);
++                if ((s.st_mode & S_IFMT) != S_IFDIR
+                  || *d->d_name != '.'
+                  || (d->d_name[1] && (d->d_name[1] != '.' || d->d_name[2])))
+                 {
+@@ -1633,7 +1639,7 @@
+     }
+ #endif
+ 
+-    utsname uts;
++    struct utsname uts;
+ 
+     if (!uname(&uts))
+     {
+@@ -1862,10 +1868,10 @@
+ 
+ fsfp_t PosixDirNotify::fsfingerprint() const
+ {
+-    struct statfs statfsbuf;
++    struct statvfs statfsbuf;
+ 
+     // FIXME: statfs() does not really do what we want.
+-    if (statfs(localbasepath.localpath.c_str(), &statfsbuf))
++    if (statvfs(localbasepath.localpath.c_str(), &statfsbuf))
+     {
+         return 0;
+     }
+@@ -1875,22 +1881,23 @@
+ 
+ bool PosixDirNotify::fsstableids() const
+ {
+-    struct statfs statfsbuf;
++    struct statvfs statfsbuf;
+ 
+-    if (statfs(localbasepath.localpath.c_str(), &statfsbuf))
++    if (statvfs(localbasepath.localpath.c_str(), &statfsbuf))
+     {
+         LOG_err << "Failed to get filesystem type. Error code: " << errno;
+         return true;
+     }
+ 
+-    LOG_info << "Filesystem type: " << statfsbuf.f_type;
++    LOG_info << "Filesystem type: " << statfsbuf.f_basetype;
+ 
+ #ifdef __APPLE__
+     return statfsbuf.f_type != 0x1c // FAT32
+         && statfsbuf.f_type != 0x1d; // exFAT
+ #else
+-    return statfsbuf.f_type != 0x4d44 // FAT
+-        && statfsbuf.f_type != 0x65735546; // FUSE
++//    return statfsbuf.f_basetype != 0x4d44 // FAT
++//        && statfsbuf.f_basetype != 0x65735546; // FUSE
++	  return true;
+ #endif
+ }
+ #endif // ENABLE_SYNC
+@@ -1927,9 +1934,9 @@
+ bool PosixFileSystemAccess::getlocalfstype(const LocalPath& path, FileSystemType& type) const
+ {
+ #if defined(__linux__) || defined(__ANDROID__)
+-    struct statfs statbuf;
++    struct statvfs statbuf;
+ 
+-    if (!statfs(path.localpath.c_str(), &statbuf))
++    if (!statvfs(path.localpath.c_str(), &statbuf))
+     {
+         switch (statbuf.f_type)
+         {
+@@ -1978,9 +1985,9 @@
+         {"ntfs",  FS_NTFS}
+     }; /* filesystemTypes */
+ 
+-    struct statfs statbuf;
++    struct statvfs statbuf;
+ 
+-    if (!statfs(path.localpath.c_str(), &statbuf))
++    if (!statvfs(path.localpath.c_str(), &statbuf))
+     {
+         auto it = filesystemTypes.find(statbuf.f_fstypename);
+ 

--- a/components/network/megasync/patches/06-TransfersSummaryWidget.cpp.patch
+++ b/components/network/megasync/patches/06-TransfersSummaryWidget.cpp.patch
@@ -1,0 +1,10 @@
+--- MEGAsync_ori/src/MEGASync/gui/TransfersSummaryWidget.cpp	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/gui/TransfersSummaryWidget.cpp	Sat Mar  5 15:32:56 2022
+@@ -7,6 +7,7 @@
+ #include <QPainterPath>
+ #include <QTimer>
+ #include "Utilities.h"
++#include "math.h"
+ 
+ TransfersSummaryWidget::TransfersSummaryWidget(QWidget *parent) :
+     QWidget(parent),

--- a/components/network/megasync/patches/07-CrashHandler.cpp.patch
+++ b/components/network/megasync/patches/07-CrashHandler.cpp.patch
@@ -1,0 +1,51 @@
+--- MEGAsync_ori/src/MEGASync/control/CrashHandler.cpp	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/control/CrashHandler.cpp	Fri Feb 18 16:33:47 2022
+@@ -12,12 +12,17 @@
+ 
+ #if defined(Q_OS_MAC)
+ #include "client/mac/handler/exception_handler.h"
+-#elif defined(Q_OS_LINUX)
+-#include "client/linux/handler/exception_handler.h"
++#elif defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
++#include "client/solaris/handler/exception_handler.h"
+ #elif defined(Q_OS_WIN32)
+ #include "client/windows/handler/exception_handler.h"
+ #endif
+ 
++#ifdef __illumos__
++#include<fcntl.h>
++#include<unistd.h>
++#endif
++
+ #ifndef WIN32
+     #ifndef CREATE_COMPATIBLE_MINIDUMPS
+ 
+@@ -26,7 +31,9 @@
+     #include <sys/utsname.h>
+ 
+ 
+-#ifdef __linux__
++
++
++#if defined(__linux__) || defined(__freebsd__) || defined(__illumos__)
+     #include <fstream>
+ 
+ string &ltrimEtcProperty(string &s, const char &c)
+@@ -310,7 +317,7 @@
+ bool DumpCallback(const wchar_t* _dump_dir,const wchar_t* _minidump_id,void* context,EXCEPTION_POINTERS* exinfo,MDRawAssertionInfo* assertion,bool success)
+ #elif defined(Q_OS_LINUX)
+ bool DumpCallback(const google_breakpad::MinidumpDescriptor &,void *context, bool success)
+-#elif defined(Q_OS_MAC)
++#elif defined(Q_OS_MAC) || defined(Q_OS_SOLARIS)
+ bool DumpCallback(const char* _dump_dir,const char* _minidump_id,void *context, bool success)
+ #endif
+ {
+@@ -347,7 +354,7 @@
+         );
+ #else
+     #ifdef CREATE_COMPATIBLE_MINIDUMPS
+-        #if defined(Q_OS_LINUX)
++        #if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+             std::string pathAsStr = dumpPath.toUtf8().constData();
+             google_breakpad::MinidumpDescriptor md(pathAsStr);
+             pHandler = new google_breakpad::ExceptionHandler(

--- a/components/network/megasync/patches/08-freeimage.cpp.patch
+++ b/components/network/megasync/patches/08-freeimage.cpp.patch
@@ -1,0 +1,26 @@
+--- MEGAsync_ori/src/MEGASync/mega/src/gfx/freeimage.cpp	Wed Mar  9 22:04:59 2022
++++ MEGAsync_patched/src/MEGASync/mega/src/gfx/freeimage.cpp	Sat Mar  5 15:49:56 2022
+@@ -38,7 +38,13 @@
+ #define OLD_FREEIMAGE
+ #endif
+ 
+-
++#include <libavformat/avformat.h>
++#include <libavcodec/avcodec.h>
++#include <libswscale/swscale.h>
++#include <libavutil/avutil.h>
++#include <libavutil/mathematics.h>
++#include <libavutil/display.h>
++#include <libavutil/imgutils.h>
+ #ifdef HAVE_FFMPEG
+ extern "C" {
+ #ifdef _WIN32
+@@ -253,7 +259,7 @@
+ 
+     // Force seeking to key frames
+     formatContext->seek2any = false;
+-    videoStream->skip_to_keyframe = true;
++//    videoStream->skip_to_keyframe = true;
+     if (decoder->capabilities & CAP_TRUNCATED)
+     {
+         codecContext->flags |= CAP_TRUNCATED;

--- a/components/network/megasync/patches/09-google_breakpad.pri.patch
+++ b/components/network/megasync/patches/09-google_breakpad.pri.patch
@@ -1,0 +1,28 @@
+--- MEGAsync_ori/src/MEGASync/google_breakpad/google_breakpad.pri	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/google_breakpad/google_breakpad.pri	Sun Mar  6 00:16:50 2022
+@@ -27,20 +27,12 @@
+ }
+ 
+ unix:!macx {
+-  SOURCES += $$PWD/client/linux/crash_generation/crash_generation_client.cc
+-  SOURCES += $$PWD/client/linux/handler/exception_handler.cc
+-  SOURCES += $$PWD/client/linux/handler/minidump_descriptor.cc
+-  SOURCES += $$PWD/client/linux/minidump_writer/minidump_writer.cc
+-  SOURCES += $$PWD/client/linux/minidump_writer/linux_dumper.cc
+-  SOURCES += $$PWD/client/linux/minidump_writer/linux_ptrace_dumper.cc
+-  SOURCES += $$PWD/client/linux/log/log.cc
++  SOURCES += $$PWD/client/solaris/handler/solaris_lwp.cc
++  SOURCES += $$PWD/client/solaris/handler/exception_handler.cc
++  SOURCES += $$PWD/client/solaris/handler/minidump_generator.cc
+   SOURCES += $$PWD/client/minidump_file_writer.cc
+-  SOURCES += $$PWD/common/linux/linux_libc_support.cc
+-  SOURCES += $$PWD/common/linux/file_id.cc
+-  SOURCES += $$PWD/common/linux/memory_mapped_file.cc
+-  SOURCES += $$PWD/common/linux/safe_readlink.cc
+-  SOURCES += $$PWD/common/linux/guid_creator.cc
+-  SOURCES += $$PWD/common/linux/elfutils.cc
++  SOURCES += $$PWD/common/solaris/file_id.cc
++  SOURCES += $$PWD/common/solaris/guid_creator.cc
+   SOURCES += $$PWD/common/string_conversion.cc
+   SOURCES += $$PWD/common/convert_UTF.c
+ 

--- a/components/network/megasync/patches/10-exception_handler.cc.patch
+++ b/components/network/megasync/patches/10-exception_handler.cc.patch
@@ -1,0 +1,10 @@
+--- MEGAsync_ori/src/MEGASync/google_breakpad/client/solaris/handler/exception_handler.cc	Wed Jan 26 20:25:17 2022
++++ MEGASync_patched/google_breakpad/client/solaris/handler/exception_handler.cc	Sat Wed  5 19:39:26 2022
+@@ -33,6 +33,7 @@
+ #include <sys/stat.h>
+ #include <sys/types.h>
+ #include <unistd.h>
++#include <limits.h>
+ 
+ #include <cassert>
+ #include <cstdlib>

--- a/components/network/megasync/patches/11-minidump_generator.cc.patch
+++ b/components/network/megasync/patches/11-minidump_generator.cc.patch
@@ -1,0 +1,95 @@
+--- MEGAsync_ori/src/MEGASync/google_breakpad/client/solaris/handler/minidump_generator.cc	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/google_breakpad/client/solaris/handler/minidump_generator.cc	Sat Mar  5 20:28:34 2022
+@@ -38,9 +38,11 @@
+ #include <unistd.h>
+ 
+ #include <cstdlib>
++#include <cstring>
+ #include <ctime>
+ 
+ #include "client/solaris/handler/minidump_generator.h"
++#include "client/solaris/handler/solaris_lwp.h"
+ #include "client/minidump_file_writer-inl.h"
+ #include "common/solaris/file_id.h"
+ 
+@@ -105,6 +107,8 @@
+   uintptr_t last_ebp = (*gregs)[R_FP];
+ #elif TARGET_CPU_X86
+   uintptr_t last_ebp = (*gregs)[EBP];
++#elif TARGET_CPU_X86_64
++  uintptr_t last_ebp = (*gregs)[EBP];
+ #endif
+   uintptr_t stack_bottom = lwp_lister->GetLwpStackBottom(last_ebp);
+   if (stack_bottom > last_ebp &&
+@@ -194,7 +198,7 @@
+ 
+   return true;
+ }
+-#elif TARGET_CPU_X86
++#elif TARGET_CPU_X86 || TARGET_CPU_X86_64
+ bool WriteContext(MDRawContextX86 *context, prgregset_t regs,
+                   prfpregset_t *fp_regs) {
+   if (!context || !regs)
+@@ -250,7 +254,7 @@
+   lwp->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextSPARC));
+   return WriteContext(context.get(), writer_args->sig_ctx);
+-#elif TARGET_CPU_X86
++#elif TARGET_CPU_X86 || TARGET_CPU_X86_64
+   UntypedMDRVA memory(minidump_writer);
+   if (!WriteLwpStack(writer_args->lwp_lister,
+                      writer_args->sig_ctx->uc_mcontext.gregs[UESP],
+@@ -264,7 +268,7 @@
+   lwp->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextX86));
+   return WriteContext(context.get(),
+-                      (int *)&writer_args->sig_ctx->uc_mcontext.gregs,
++                      writer_args->sig_ctx->uc_mcontext.gregs,
+                       &writer_args->sig_ctx->uc_mcontext.fpregs);
+ #endif
+ }
+@@ -290,7 +294,7 @@
+   lwp->thread_id = lsp->pr_lwpid;
+   lwp->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextSPARC));
+-#elif TARGET_CPU_X86
++#elif TARGET_CPU_X86 || TARGET_CPU_X86_64
+   if (!WriteLwpStack(lwp_lister,
+                      (*gregs)[UESP],
+                      &memory,
+@@ -306,7 +310,7 @@
+   lwp->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextX86));
+ #endif /* TARGET_CPU_XXX */
+-  return WriteContext(context.get(), (int *)gregs, &fp_regs);
++  return WriteContext(context.get(), (greg_t*)&gregs, &fp_regs);
+ }
+ 
+ bool WriteCPUInformation(MDRawSystemInfo *sys_info) {
+@@ -605,7 +609,7 @@
+   exception.get()->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextSPARC));
+   return WriteContext(context.get(), writer_args->sig_ctx);
+-#elif TARGET_CPU_X86
++#elif TARGET_CPU_X86 || TARGET_CPU_X86_64
+   if (writer_args->sig_ctx != NULL) {
+     exception.get()->exception_record.exception_address =
+       writer_args->sig_ctx->uc_mcontext.gregs[EIP];
+@@ -620,7 +624,7 @@
+   exception.get()->thread_context = context.location();
+   memset(context.get(), 0, sizeof(MDRawContextX86));
+   return WriteContext(context.get(),
+-                      (int *)&writer_args->sig_ctx->uc_mcontext.gregs,
++                      writer_args->sig_ctx->uc_mcontext.gregs,
+                       NULL);
+ #endif
+ }
+@@ -702,6 +706,8 @@
+           writer_args->sig_ctx->uc_mcontext.gregs[REG_O6]
+ #elif TARGET_CPU_X86
+           writer_args->sig_ctx->uc_mcontext.gregs[UESP]
++#elif TARGET_CPU_X86_64
++		  writer_args->sig_ctx->uc_mcontext.gregs[UESP]
+ #endif
+       );
+ 

--- a/components/network/megasync/patches/12-solaris_lwp.cc.patch
+++ b/components/network/megasync/patches/12-solaris_lwp.cc.patch
@@ -1,0 +1,10 @@
+--- MEGAsync_ori/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.cc	Sat Mar  5 19:57:17 2022
+@@ -44,6 +44,7 @@
+ #include <cassert>
+ #include <cstdio>
+ #include <cstdlib>
++#include <cstring>
+ #include <functional>
+ 
+ #include "client/solaris/handler/solaris_lwp.h"

--- a/components/network/megasync/patches/13-solaris_lwp.h.patch
+++ b/components/network/megasync/patches/13-solaris_lwp.h.patch
@@ -1,0 +1,11 @@
+--- MEGAsync_ori/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.h	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/google_breakpad/client/solaris/handler/solaris_lwp.h	Sat Mar  5 15:56:15 2022
+@@ -36,6 +36,8 @@
+ #define TARGET_CPU_SPARC 1
+ #elif defined(i386) || defined(__i386)
+ #define TARGET_CPU_X86 1
++#elif defined(amd64) || defined(__amd64)
++#define TARGET_CPU_X86_64 1
+ #else
+ #error "cannot determine cpu type"
+ #endif

--- a/components/network/megasync/patches/14-main.cpp.patch
+++ b/components/network/megasync/patches/14-main.cpp.patch
@@ -1,0 +1,101 @@
+--- MEGAsync_ori/src/MEGASync/main.cpp	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/main.cpp	Sun Mar  6 00:21:32 2022
+@@ -9,7 +9,7 @@
+ #include <QFontDatabase>
+ #include <assert.h>
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     #include <signal.h>
+     #include <condition_variable>
+ #endif
+@@ -24,7 +24,7 @@
+ #include <Shellapi.h>
+ #endif
+ 
+-#if defined(WIN32) || defined(Q_OS_LINUX)
++#if defined(WIN32) || defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+ #include <QScreen>
+ #endif
+ 
+@@ -59,7 +59,7 @@
+ }
+ 
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+ MegaApplication *theapp = NULL;
+ bool waitForRestartSignal = false;
+ std::mutex mtxcondvar;
+@@ -229,7 +229,7 @@
+         return 0;
+     }
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+ 
+     // Ensure interesting signals are unblocked.
+     sigset_t signalstounblock;
+@@ -315,7 +315,7 @@
+ 
+ #endif
+ 
+-#if defined(Q_OS_LINUX)
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     if (!(getenv("DO_NOT_SET_QT_PLUGIN_PATH")))
+     {
+         if (QDir(QString::fromUtf8("/opt/mega/plugins")).exists())
+@@ -333,7 +333,7 @@
+     }
+ #endif
+ 
+-#if defined(Q_OS_LINUX) && QT_VERSION >= 0x050C00
++#if (defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)) && QT_VERSION >= 0x050C00
+     // Linux && Qt >= 5.12.0
+     if (!(getenv("DO_NOT_UNSET_XDG_SESSION_TYPE")))
+     {
+@@ -350,7 +350,7 @@
+     ScaleFactorManager scaleFactorManager(OsType::WIN);
+ #endif
+ 
+-#if defined(Q_OS_LINUX)
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     ScaleFactorManager scaleFactorManager(OsType::LINUX);
+ #endif
+ 
+@@ -364,7 +364,7 @@
+     }
+ #endif
+ 
+-#if defined(Q_OS_LINUX)
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+ #if QT_VERSION >= 0x050000
+     if (!(getenv("DO_NOT_UNSET_QT_QPA_PLATFORMTHEME")) && getenv("QT_QPA_PLATFORMTHEME"))
+     {
+@@ -389,7 +389,7 @@
+ 
+ 
+     MegaApplication app(argc, argv);
+-#if defined(Q_OS_LINUX)
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     theapp = &app;
+     appToWaitForSignal = QString::fromUtf8("\"%1\"").arg(MegaApplication::applicationFilePath());
+     for (int i = 1; i < argc; i++)
+@@ -413,7 +413,7 @@
+     }
+ #endif
+ 
+-#if defined(Q_OS_LINUX) && QT_VERSION >= 0x050600
++#if (defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)) && QT_VERSION >= 0x050600
+     for (const auto& screen : app.screens())
+     {
+         MegaApi::log(MegaApi::LOG_LEVEL_INFO, ("Device pixel ratio on '" +
+@@ -560,7 +560,7 @@
+     int toret = app.exec();
+ 
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     theapp = nullptr;
+ #endif
+     return toret;

--- a/components/network/megasync/patches/15-MegaApplication.cpp.patch
+++ b/components/network/megasync/patches/15-MegaApplication.cpp.patch
@@ -1,0 +1,74 @@
+--- MEGAsync_ori/src/MEGASync/MegaApplication.cpp	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/MegaApplication.cpp	Sun Mar  6 01:01:28 2022
+@@ -22,7 +22,7 @@
+ 
+ #include <assert.h>
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     #include <signal.h>
+     #include <condition_variable>
+     #include <QSvgRenderer>
+@@ -54,7 +54,7 @@
+ 
+ void MegaApplication::loadDataPath()
+ {
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     dataPath = QStandardPaths::writableLocation(QStandardPaths::GenericDataLocation)
+                + QString::fromUtf8("/data/Mega Limited/MEGAsync");
+ #else
+@@ -89,7 +89,7 @@
+     logToStdout = true;
+ #endif
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     if (argc == 2)
+     {
+          if (!strcmp("--debug", argv[1]))
+@@ -418,7 +418,7 @@
+     indexing = false;
+     setQuitOnLastWindowClosed(false);
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     isLinux = true;
+ #else
+     isLinux = false;
+@@ -737,7 +737,7 @@
+     createTrayIcon();
+ }
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+ void MegaApplication::setTrayIconFromTheme(QString icon)
+ {
+     QString name = QString(icon).replace(QString::fromUtf8("://images/"), QString::fromUtf8("mega")).replace(QString::fromUtf8(".svg"),QString::fromUtf8(""));
+@@ -965,7 +965,7 @@
+ 
+ void MegaApplication::start()
+ {
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     QSvgRenderer qsr; //to have svg library linked
+ #endif
+ 
+@@ -2250,7 +2250,7 @@
+ 
+     if (trayIcon)
+     {
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+         if (counter==4 && getenv("XDG_CURRENT_DESKTOP") && !strcmp(getenv("XDG_CURRENT_DESKTOP"),"XFCE"))
+         {
+             trayIcon->hide();
+@@ -5793,7 +5793,7 @@
+         whyAmIBlocked();
+     }
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     if (getenv("XDG_CURRENT_DESKTOP") && (
+                 !strcmp(getenv("XDG_CURRENT_DESKTOP"),"ubuntu:GNOME")
+                 || !strcmp(getenv("XDG_CURRENT_DESKTOP"),"LXDE")

--- a/components/network/megasync/patches/16-MegaApplication.h.patch
+++ b/components/network/megasync/patches/16-MegaApplication.h.patch
@@ -1,0 +1,11 @@
+--- MEGAsync_ori/src/MEGASync/MegaApplication.h	Wed Jan 26 20:25:17 2022
++++ MEGAsync_patched/src/MEGASync/MegaApplication.h	Sun Mar  6 00:22:00 2022
+@@ -100,7 +100,7 @@
+ {
+     Q_OBJECT
+ 
+-#ifdef Q_OS_LINUX
++#if defined(Q_OS_LINUX) || defined(Q_OS_SOLARIS)
+     void setTrayIconFromTheme(QString icon);
+ #endif
+ 

--- a/components/network/megasync/pkg5
+++ b/components/network/megasync/pkg5
@@ -1,0 +1,28 @@
+{
+    "dependencies": [
+        "SUNWcs",
+        "database/sqlite-3",
+        "developer/versioning/git",
+        "library/cryptopp",
+        "library/libcares",
+        "library/libfreeimage",
+        "library/libmediainfo",
+        "library/libuv",
+        "library/libzen",
+        "library/qt5",
+        "library/security/libsodium",
+        "library/security/openssl",
+        "library/zlib",
+        "shell/ksh93",
+        "system/library",
+        "system/library/g++-7-runtime",
+        "system/library/gcc-7-runtime",
+        "system/library/math",
+        "web/curl",
+        "x11/library/libxcb"
+    ],
+    "fmris": [
+        "network/megasync"
+    ],
+    "name": "MEGAsync"
+}

